plugin/src/ios/CDVMParticle.m

```diff
@@ -530,5 +581,37 @@ + (MPIdentity)ConvertIdentityType:(NSString *)val {
     }
 }
 
++ (MPGDPRConsent *)MPGDPRConsent:(id)json {
+    NSString *dateString =  json[@"timestamp"];
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    [dateFormatter setDateFormat:@"yyyy-MM-dd HH:mm:ss"];
+    NSDate *timestamp = [dateFormatter dateFromString:dateString];
+    
+    MPGDPRConsent *consentState = [[MPGDPRConsent alloc] init];
+    consentState.consented = json[@"consented"];
```

Comment:
This is trying to set to a BOOL from an object, maybe you need to call 'boolValue'?